internal/saucecloud/cloud.go

```diff
@@ -336,11 +340,31 @@ func (r *CloudRunner) runJobs(jobOpts chan job.StartOptions, results chan<- resu
 			continue
 		}
 
+		// Attempt starts from 0 while MaxAttempts starts from 1. So here is comparing Attempt with MaxAttempts-1
+		if opts.Attempt < opts.MaxAttempts-1 && opts.CurrentPassCount < opts.PassCount {
+			opts.Attempt++
+			jobOpts <- opts
+			log.Info().Str("suite", opts.DisplayName).Str("attempt", fmt.Sprintf("%d of %d", opts.Attempt+1, opts.MaxAttempts)).Msg("Retrying suite.")
+			continue
+		}
+
 		if r.FailFast && !jobData.Passed {
 			log.Warn().Err(err).Msg("FailFast mode enabled. Skipping upcoming suites.")
 			r.interrupted = true
 		}
 
+		if opts.MaxAttempts != 0 {
```

Comment:
A bit confusing if this statement can ever be false in most circumstances, since the default min value is 1.

Reply:
Will remove this check.